addons/crm/models/crm_team.py

```diff
@@ -113,12 +146,329 @@ def _alias_get_creation_values(self):
         if self.id:
             if not self.use_leads and not self.use_opportunities:
                 values['alias_name'] = False
-            values['alias_defaults'] = defaults = ast.literal_eval(self.alias_defaults or "{}")
+            values['alias_defaults'] = defaults = literal_eval(self.alias_defaults or "{}")
             has_group_use_lead = self.env.user.has_group('crm.group_use_lead')
             defaults['type'] = 'lead' if has_group_use_lead and self.use_leads else 'opportunity'
             defaults['team_id'] = self.id
         return values
 
+    # ------------------------------------------------------------
+    # LEAD ASSIGNMENT
+    # ------------------------------------------------------------
+
+    @api.model
+    def _cron_assign_leads(self):
+        """ Cron method assigning leads. It is based on cron configuration to
+        know parameters of assignment to perform. """
+        assign_cron = self.sudo().env.ref('crm.ir_cron_crm_lead_assign', raise_if_not_found=False)
+        work_days = 2
+        if assign_cron and assign_cron.active:
+            if assign_cron.interval_type == 'months':
+                work_days = 2 * assign_cron.interval_number * 30
+            elif assign_cron.interval_type == 'weeks':
+                work_days = 2 * assign_cron.interval_number * 7
+            elif assign_cron.interval_type == 'days':
+                work_days = 2 * assign_cron.interval_number * 1
+        self.env['crm.team'].search([('assignment_optout', '=', False)])._action_assign_leads(work_days=work_days)
+        return True
+
+    def action_assign_leads(self, work_days=2):
+        """ Manual (direct) leads assignment. This method both
+
+          * assigns leads to teams given by self;
+          * assigns leads to salespersons belonging to self;
+
+        See sub methods for more details about assign process.
+
+        :param int work_days: number of work days to consider when assigning leads
+          to teams or salespersons. We consider that Member.assignment_max (or
+          its equivalent on team model) targets 30 work days. We make a ratio
+          between expected number of work days and maximum assignment for those
+          30 days to know lead count to assign.
+
+        :return action: a client notification giving some insights on assign
+          process;
+        """
+        teams_data, members_data = self._action_assign_leads(work_days=work_days)
+
+        # format result messages
+        notif_message = self._action_assign_leads_log(teams_data, members_data, html=False)
+        html_message = self._action_assign_leads_log(teams_data, members_data, html=True)
+
+        # log a note in case of manual assign
+        log_action = _("Lead Assignment done by %(user_name)s", user_name=self.env.user.name)
+        log_message = "<p>%s<br /><br />%s</p>" % (log_action, html_message)
+        self._message_log_batch(bodies=dict((team.id, log_message) for team in self))
```

Comment:
So it's the same message on every team?
Shouldn't the message only contain leads assigned within that specific team?

Maybe it's a bit too complicated to handle / not necessary?

Reply:
Manual assign is normally done one team at a time as it is an action button. Automatic assign doing it on all teams will not log anything (too verbose, noisy). So I voted for a simple multi-enabled method even if its use is better fit for record set containing a single team indeed.